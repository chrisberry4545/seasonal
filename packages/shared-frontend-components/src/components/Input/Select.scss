@import '../../styles/settings';
@import '../../styles/tools';

.c-select {
  border: $input-border;
  border-radius: $input-border-radius;
  color: $primary-text-color;
  background: none;
  height: 2.375rem;
}
