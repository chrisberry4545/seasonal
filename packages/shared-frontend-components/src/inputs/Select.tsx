import React, { FC } from 'react';
import { ISelectOption } from './select-option.interface';

export const Select: FC<{
  className?: string,
  onChange: (newValue: string) => void,
  options: ISelectOption[],
  value: string | number
}> = ({
  className,
  onChange,
  options,
  value
}) => (
  <select
    className={`${(className || '')}`}
    value={value}
    onChange={(event) => onChange(event.target.value)}>
      {
        options.map((option) =>
          <option key={option.value} value={option.value}>{option.label}</option>
        )
      }
  </select>
);
