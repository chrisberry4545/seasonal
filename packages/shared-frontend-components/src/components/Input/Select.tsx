import React, { FC } from 'react';
import { ISelectOption } from './select-option.interface';
import './Select.scss';

export const Select: FC<{
  className?: string,
  onChange: (newValue: string) => void,
  options: ISelectOption[],
  value: string | number
}> = ({
  className,
  onChange,
  options,
  value
}) => (
  <select
    className={`${(className || '')} c-select`}
    value={value}
    onChange={(event) => onChange(event.target.value)}>
      {
        options.map((option) =>
          <option key={option.value} value={option.value}>{option.label}</option>
        )
      }
  </select>
);
