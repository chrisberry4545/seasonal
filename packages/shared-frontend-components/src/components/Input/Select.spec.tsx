import React from 'react';
import { Select } from './Select';
import { shallow, ShallowWrapper } from 'enzyme';
import { ISelectOption } from './select-option.interface';

describe('<Select />', () => {
  let wrapper: ShallowWrapper;
  let mockOnChange: jest.Mock;
  const options = [
    {
      label: 'l1',
      value: 'v1'
    },
    {
      label: 'l2',
      value: 'v2'
    }
  ] as ISelectOption[];

  beforeEach(() => {
    mockOnChange = jest.fn();
    wrapper = shallow(
      <Select
        onChange={mockOnChange}
        className='class'
        options={options}
        value={options[0].value} />
    );
  });

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());

  test('outputs the value when changed', () => {
    wrapper.find('select').simulate('change', { target: { value: 'new-val' } });
    expect(mockOnChange).toHaveBeenCalledWith('new-val');
  });
});
