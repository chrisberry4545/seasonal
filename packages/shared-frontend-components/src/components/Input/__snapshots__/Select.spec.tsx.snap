// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders correctly 1`] = `
<select
  className="class c-select"
  onChange={[Function]}
  value="v1"
>
  <option
    key="v1"
    value="v1"
  >
    l1
  </option>
  <option
    key="v2"
    value="v2"
  >
    l2
  </option>
</select>
`;
